src/components/AssetManagementPanel/AssetManagementForm/FormVariants/AcceptSurrenderForm/AcceptSurrenderForm.tsx

```diff
@@ -21,27 +24,56 @@ interface AcceptSurrenderFormProps {
   acceptSurrenderingState: string;
   setFormActionNone: () => void;
   setShowEndorsementChain: (payload: boolean) => void;
+  handleRejectSurrender: (lastBeneficary: string) => void;
+  rejectSurrenderingState: string;
 }
 
 export const AcceptSurrenderForm = ({
+  tokenId,
   formAction,
   tokenRegistryAddress,
   handleAcceptSurrender,
   acceptSurrenderingState,
   setFormActionNone,
   setShowEndorsementChain,
+  handleRejectSurrender,
+  rejectSurrenderingState,
 }: AcceptSurrenderFormProps) => {
-  const isPendingConfirmation = acceptSurrenderingState === FormState.PENDING_CONFIRMATION;
-  const isConfirmed = acceptSurrenderingState === FormState.CONFIRMED;
-
   const { showOverlay } = useContext(OverlayContext);
+  const { endorsementChain } = useEndorsementChain(tokenRegistryAddress, tokenId);
+
+  const isAcceptSurrenderingPendingConfirmation = acceptSurrenderingState === FormState.PENDING_CONFIRMATION;
+  const isRejectSurrenderingPendingConfirmation = rejectSurrenderingState === FormState.PENDING_CONFIRMATION;
+  const isPendingConfirmation = isAcceptSurrenderingPendingConfirmation || isRejectSurrenderingPendingConfirmation;
+  const isAcceptSurrenderConfirmed = acceptSurrenderingState === FormState.CONFIRMED;
+  const isRejectSurrenderConfirmed = rejectSurrenderingState === FormState.CONFIRMED;
+
+  const lastTransferEvent = endorsementChain
+    ?.filter(({ eventType }) => eventType === "Transfer")
+    .pop() as TitleEscrowEvent;
+  const lastBeneficary = lastTransferEvent?.beneficiary;
+
+  const onClickRejectSurrender = () => {
+    showOverlay(
+      showDocumentTransferMessage(MessageTitle.CONFIRM_REJECT_SURRENDER_DOCUMENT, {
+        isSuccess: true,
+        beneficiaryAddress: lastBeneficary,
+        isConfirmationMessage: true,
+        onConfirmaionAction: () => handleRejectSurrender(lastBeneficary),
+      })
+    );
+  };
 
   useEffect(() => {
-    if (isConfirmed) {
+    if (isAcceptSurrenderConfirmed) {
       showOverlay(showDocumentTransferMessage(MessageTitle.ACCEPT_SURRENDER_DOCUMENT, { isSuccess: true }));
       setFormActionNone();
     }
-  }, [isConfirmed, showOverlay, setFormActionNone]);
+    if (isRejectSurrenderConfirmed) {
+      showOverlay(showDocumentTransferMessage(MessageTitle.REJECT_SURRENDER_DOCUMENT, { isSuccess: false }));
+      setFormActionNone();
+    }
+  }, [isAcceptSurrenderConfirmed, showOverlay, setFormActionNone, isRejectSurrenderConfirmed]);
```

Comment:
Should they be in separate useEffect so that change in isRejectSurrenderConfirmed should not affect isAcceptSurrenderConfirmed, although they are mutually exclusive...